packages/core/plugin/src/withSentryAndroidGradlePlugin.ts

```diff
@@ -0,0 +1,96 @@
+import { withAppBuildGradle, withProjectBuildGradle } from '@expo/config-plugins';
+
+export interface SentryAndroidGradlePluginOptions {
+  sentryAndroidGradlePluginVersion?: string;
+  includeProguardMapping?: boolean;
+  dexguardEnabled?: boolean;
+  autoUploadNativeSymbols?: boolean;
+  autoUploadProguardMapping?: boolean;
+  uploadNativeSymbols?: boolean;
+  includeNativeSources?: boolean;
+  includeSourceContext?: boolean;
+  autoInstallationEnabled?: boolean;
+}
+
+/**
+ * Adds the Sentry Android Gradle Plugin to the project.
+ * https://docs.sentry.io/platforms/react-native/manual-setup/manual-setup/#enable-sentry-agp
+ */
+export function withSentryAndroidGradlePlugin(config: any, options: SentryAndroidGradlePluginOptions = {}): any {
+  const version = options.sentryAndroidGradlePluginVersion;
+  const includeProguardMapping = options.includeProguardMapping ?? true;
+  const dexguardEnabled = options.dexguardEnabled ?? false;
+  const autoUploadProguardMapping = options.autoUploadProguardMapping ?? true;
+  const uploadNativeSymbols = options.uploadNativeSymbols ?? true;
+  const autoUploadNativeSymbols = options.autoUploadNativeSymbols ?? true;
+  const includeNativeSources = options.includeNativeSources ?? true;
+  const includeSourceContext = options.includeSourceContext ?? false;
+  const autoInstallationEnabled = options.autoInstallationEnabled ?? false;
+
+  // Modify android/build.gradle
+  const withSentryProjectBuildGradle = (config: any): any => {
+    return withProjectBuildGradle(config, (projectBuildGradle: any) => {
```

Comment:
Should we also check if the lang is `groovy`?

Reply:
Good point 👍 Added a check with 9d6d9d204f32a6bc35d8bd834c0e1b023642e338